store/src/java/com/zimbra/cs/redolog/RedoLogManager.java

```diff
@@ -210,11 +218,20 @@ public synchronized void start() {
         }
 
         long fsyncInterval = RedoConfig.redoLogFsyncIntervalMS();
-        mLogWriter = createLogWriter(this, mLogFile, fsyncInterval);
+        if (useDbAsLogStorage){
+            mLogWriter = createLogWriter(this);
+        } else {
+            mLogWriter = mLogWriter = createLogWriter(this, mLogFile, fsyncInterval);
+        }
```

Comment:
Replace this with `mLogWriter = new DbLogWriter(this);`. Synacor can update to switch using dependency injection or configuration.

If dependency injection is eventually used the constructor for `DbLogWriter` and `FileLogWriter` should be updated to read whatever they need from the `RedoLogManager` passed to their constructor.

`FileLogWriter` constructor could be refactored to match `DbLogWriter` easily. We do not need to do this as part of this PR.

```java
    public FileLogWriter(RedoLogManager redoLogMgr) {
        mRedoLogMgr = redoLogMgr;

        mHeader = new FileHeader(sServerId);
        mFile = redoLogMgr.getLogFile();
        mFileSize = mFile.length();
        mLastLogTime = mFile.lastModified();

        mFsyncIntervalMS = RedoConfig.redoLogFsyncIntervalMS();
        mFsyncDisabled = DebugConfig.disableRedoLogFsync;

        mFsyncCount = mLogCount = 0;

        mCommitNotifyQueue = new CommitNotifyQueue(100);
    }
```